tsdb/wal/wal.go

```diff
@@ -258,13 +264,14 @@ func NewSize(logger log.Logger, reg prometheus.Registerer, dir string, segmentSi
 		stopc:       make(chan chan struct{}),
 		compress:    compress,
 	}
-	w.metrics = newWALMetrics(w, reg)
 
 	_, last, err := w.Segments()
 	if err != nil {
-		return nil, errors.Wrap(err, "get segment range")
+		return w, errors.Wrap(err, "get segment range")
```

Comment:
we need a directory path in order to repair/delete the corrupted segments which is done inside `repairUnsequentialSegments()`. If I return nil, then I might not be able to pass the directory since the `Repair()` calls the `repairUnsequentialSegments()`. Hence I use the interface to pass on the path of the directory.